analyzer/src/main/kotlin/managers/Stack.kt

```diff
@@ -0,0 +1,157 @@
+
+            val dependencies = mutableMapOf<String, MutableList<String>>()
+
+            dotParser.edges.values.forEach { edge ->
+                val parent = edge.node1.id.removeSurrounding("\"")
+                val child = edge.node2.id.removeSurrounding("\"")
+                dependencies.getOrPut(parent) { mutableListOf() } += child
+            }
+
+            return dependencies
+        }
+
+        fun mapNamesToVersions(scope: String): Map<String, String> {
+            val dependencies = runStack("ls", "dependencies", "--$scope").stdout()
+            return dependencies.lines().associate {
+                Pair(it.substringBefore(" "), it.substringAfter(" "))
+            }
+        }
+
+        val allPackages = sortedSetOf<Package>()
+
+        val externalChildren = mapParentsToChildren("external")
+        val externalVersions = mapNamesToVersions("external")
+        val externalDependencies = sortedSetOf<PackageReference>()
+        buildDependencyTree(projectName, externalChildren, externalVersions, allPackages, externalDependencies)
+
+        val testChildren = mapParentsToChildren("test")
+        val testVersions = mapNamesToVersions("test")
+        val testDependencies = sortedSetOf<PackageReference>()
+        buildDependencyTree(projectName, testChildren, testVersions, allPackages, testDependencies)
+
+        val benchChildren = mapParentsToChildren("bench")
+        val benchVersions = mapNamesToVersions("bench")
+        val benchDependencies = sortedSetOf<PackageReference>()
+        buildDependencyTree(projectName, benchChildren, benchVersions, allPackages, benchDependencies)
+
+        val scopes = sortedSetOf(
+                Scope("external", true, externalDependencies),
+                Scope("test", false, testDependencies),
+                Scope("bench", false, benchDependencies)
+        )
+
+        val project = Project(
+                id = Identifier(
+                        provider = "Hackage",
+                        namespace = "",
+                        name = projectName,
+                        version = ""
+                ),
+                definitionFilePath = VersionControlSystem.getPathToRoot(definitionFile) ?: "",
+                declaredLicenses = sortedSetOf(),
+                vcs = VcsInfo.EMPTY,
+                vcsProcessed = processProjectVcs(workingDir, homepageUrl = ""),
+                homepageUrl = "",
+                scopes = scopes
+        )
+
+        // Stack does not support lock files, so hard-code "allowDynamicVersions" to "true".
+        return ProjectAnalyzerResult(true, project,
+                allPackages.map { it.toCuratedPackage() }.toSortedSet())
+    }
+
+    private fun buildDependencyTree(parentName: String,
+                                    childMap: Map<String, List<String>>, versionMap: Map<String, String>,
+                                    allPackages: SortedSet<Package>, dependencies: SortedSet<PackageReference>) {
```

Comment:
Should we start using this proposed formatting?
https://kotlinlang.org/docs/reference/coding-conventions.html#function-formatting

Reply:
I basically have nothing against that style, it's just not how the IDE formats the code by default, which makes it a bit inconvenient to use.

When using that style, do you put *each* argument on a single line, even if multiple shorter arguments would fit into a single line?

Reply:
Yes, one argument per line. But we haven't used this style so far, so we could also refactor the code later if we want to use it.